### PR TITLE
feat: datagrid keyboard shortcuts

### PR DESCRIPTION
**Pull request by thetaungg**
- [x] Keyboard navigation for Focus Cell. Arrow keys to change focus cell
- [x] Tab and Shift + Tab to move forward and backward horizontally to change focus cell
- [x] Cmd/Ctrl + Arrow keys to scroll to the edge of the row or column and focus the available cell
- [ ] Shift + Arrow keys - Select cells adjacent to the focused cell
- [ ] Cmd/Ctrl + Shift + Arrow keys - Select cells till the end of the direction of the keys

### CODE REVIEW

**Review by elvisduru — APPROVED** [review]

File: src/datagrid/plugins/cell-focus/useSetFocusCellByDirection.tsx [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+            if (isNil(rowIndex) || isNil(columnIndex)) return;
+
+            const { columns } = dataTableStoreRef.current;
+            const { records } = tableManagerStore.current;
+
+            let newRowIndex = rowIndex;
+            let newColumnIndex = columnIndex;
+
+            if (direction === 'up' || direction === 'down') {
```
Comment (shashank1010) [inline src/datagrid/plugins/cell-focus/useSetFocusCellByDirection.tsx:48]: quick fix to replace this with switch

File: src/datagrid/shortcuts.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import type { Shortcut } from '@bambooapp/bamboo-molecules/shortcuts-manager';
+import { isMac } from '@bambooapp/bamboo-molecules';
+
+export const shortcuts: Shortcut[] = [
+    {
+        name: 'move-cell-focus',
+        keys: [
```
Comment (shashank1010) [inline src/datagrid/shortcuts.ts:7]: can we accept `(string | string[])[]` and normalise it with code?

**Review by shashank1010 — CHANGES_REQUESTED** [review]

File: src/datagrid/plugins/cell-focus/useSetFocusCellByDirection.tsx (hunk 1 above)
Comment (thetaungg) [reply src/datagrid/plugins/cell-focus/useSetFocusCellByDirection.tsx]: done

File: src/datagrid/shortcuts.ts (hunk 2 above)
Comment (thetaungg) [reply src/datagrid/shortcuts.ts:7]: done

**Review by shashank1010 — APPROVED** [review]

**Pull request merged**